From d15f4477ca2f199b78dcec69a948f395930d6ec9 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 20 Oct 2022 18:46:48 +0200
Subject: [PATCH 30/41] Revert "blk-wbt: don't show valid wbt_lat_usec in sysfs
 while wbt is disabled"

This reverts commit 9c349c31cd2985dcc940eaf8d6925f0100437ef2.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/blk-sysfs.c | 6 +-----
 block/blk-wbt.c   | 8 --------
 block/blk-wbt.h   | 5 -----
 3 files changed, 1 insertion(+), 18 deletions(-)

diff --git a/block/blk-sysfs.c b/block/blk-sysfs.c
index 1df3e22e1..e1f009aba 100644
--- a/block/blk-sysfs.c
+++ b/block/blk-sysfs.c
@@ -467,14 +467,10 @@ static ssize_t queue_io_timeout_store(struct request_queue *q, const char *page,
 
 static ssize_t queue_wb_lat_show(struct request_queue *q, char *page)
 {
-	u64 lat;
-
 	if (!wbt_rq_qos(q))
 		return -EINVAL;
 
-	lat = wbt_disabled(q) ? 0 : div_u64(wbt_get_min_lat(q), 1000);
-
-	return sprintf(page, "%llu\n", lat);
+	return sprintf(page, "%llu\n", div_u64(wbt_get_min_lat(q), 1000));
 }
 
 static ssize_t queue_wb_lat_store(struct request_queue *q, const char *page,
diff --git a/block/blk-wbt.c b/block/blk-wbt.c
index 07ed0b0ae..4680691a9 100644
--- a/block/blk-wbt.c
+++ b/block/blk-wbt.c
@@ -422,14 +422,6 @@ static void wbt_update_limits(struct rq_wb *rwb)
 	rwb_wake_all(rwb);
 }
 
-bool wbt_disabled(struct request_queue *q)
-{
-	struct rq_qos *rqos = wbt_rq_qos(q);
-
-	return !rqos || RQWB(rqos)->enable_state == WBT_STATE_OFF_DEFAULT ||
-	       RQWB(rqos)->enable_state == WBT_STATE_OFF_MANUAL;
-}
-
 u64 wbt_get_min_lat(struct request_queue *q)
 {
 	struct rq_qos *rqos = wbt_rq_qos(q);
diff --git a/block/blk-wbt.h b/block/blk-wbt.h
index e3ea6e7e2..7fe98638f 100644
--- a/block/blk-wbt.h
+++ b/block/blk-wbt.h
@@ -96,7 +96,6 @@ void wbt_enable_default(struct request_queue *);
 
 u64 wbt_get_min_lat(struct request_queue *q);
 void wbt_set_min_lat(struct request_queue *q, u64 val);
-bool wbt_disabled(struct request_queue *);
 
 void wbt_set_write_cache(struct request_queue *, bool);
 
@@ -128,10 +127,6 @@ static inline u64 wbt_default_latency_nsec(struct request_queue *q)
 {
 	return 0;
 }
-static inline bool wbt_disabled(struct request_queue *q)
-{
-	return true;
-}
 
 #endif /* CONFIG_BLK_WBT */
 
-- 
2.38.0.rc1.6.g4fd6c5e444

